spec/requests/storage_locations_requests_spec.rb

```diff
@@ -65,6 +65,93 @@
+                item3.id => 1
+              },
+              storage_location_with_duplicate_item.id => {
+                item3.id => 1
+              },
+              storage_location_with_unique_item.id => {
+                item4.id => 5
+              }
+            })
+          end
+
+          it "generates header with Storage Location fields followed by alphabetized item names" do
+            get storage_locations_path(default_params.merge(format: response_format))
+            expect(response.body.split("\n")[0]).to eq([StorageLocation.csv_export_headers, item1.name, item2.name, item3.name, item4.name].join(','))
+          end
+
+          it "generates data row for storage location with 1 of each of 3 items, and 0 of the last item" do
+            get storage_locations_path(default_params.merge(format: response_format))
+            csv_rows = response.body.split("\n")
+            storage_location_row = csv_rows.find { |line| line.start_with?(storage_location_with_items.name) }
+
+            expect(storage_location_row).to eq([
+              storage_location_with_items.name,
+              "\"" + storage_location_with_items.address + "\"",
+              storage_location_with_items.square_footage,
+              storage_location_with_items.warehouse_type,
+              3,
+              1,
+              1,
+              1,
+              0
+            ].join(","))
+          end
+
+          it "generates data row for storage location with duplicate item" do
+            get storage_locations_path(default_params.merge(format: response_format))
+            csv_rows = response.body.split("\n")
+            storage_location_row = csv_rows.find { |line| line.start_with?(storage_location_with_duplicate_item.name) }
+
+            expect(storage_location_row).to eq([
+              storage_location_with_duplicate_item.name,
+              "\"" + storage_location_with_duplicate_item.address + "\"",
+              storage_location_with_duplicate_item.square_footage,
+              storage_location_with_duplicate_item.warehouse_type,
+              1,
+              0,
+              0,
+              1,
+              0
+            ].join(","))
+          end
+
+          it "generates data row for storage location with unique item" do
+            get storage_locations_path(default_params.merge(format: response_format))
+            csv_rows = response.body.split("\n")
+            storage_location_row = csv_rows.find { |line| line.start_with?(storage_location_with_unique_item.name) }
+
+            expect(storage_location_row).to eq([
+              storage_location_with_unique_item.name,
+              "\"" + storage_location_with_unique_item.address + "\"",
+              storage_location_with_unique_item.square_footage,
+              storage_location_with_unique_item.warehouse_type,
+              5,
+              0,
+              0,
+              0,
+              5
+            ].join(","))
+          end
+        end
```

Comment:
I can update the tests to use this technique (actually with this, can probably get all of it into just one test).

Only tricky thing could be one of the csv rows is for a storage location that comes from either seeds or fixtures, not data explicitly generated by the test. For example, the `Smithsonian Conservation Center` is not from the test:
```
Name,Address,Square Footage,Warehouse Type,Total Inventory,A,B,C,D
Smithsonian Conservation Center,"1500 Remount Road, Front Royal, VA 22630",100,Residential space used,0,0,0,0,0
Storage Location with Duplicate Items,"1500 Remount Road, Front Royal, VA 22630",100,Residential space used,1,0,0,1,0
Storage Location with Items,"1500 Remount Road, Front Royal, VA 22630",100,Residential space used,3,1,1,1,0
Storage Location with Unique Items,"1500 Remount Road, Front Royal, VA 22630",100,Residential space used,5,0,0,0,5
```

I can add an assertion for it, but if the seeds/fixtures should change in the future, this test would break.

Thoughts on how to deal with this?

Reply:
Another observation: That storage location that's not generated explicitly by the test uses the `StorageLocation::WAREHOUSE_TYPES.sample` for warehouse_type, resulting in a different csv output each time which would make this test flaky.

I actually ran into that earlier with my own factory created storage locations, which I fixed by always selecting a specific warehouse type when creating, eg:
```ruby
let(:storage_location_with_items) { create(:storage_location, name: "Storage Location with Items", warehouse_type: StorageLocation::WAREHOUSE_TYPES.first) }
```

Potential Solutions:
1. Figure out where that additional storage location is coming from and try to remove it from fixtures or seeds (but this might break other tests).
2. Process the `response.body` to remove that row before using it in test expectation. But it might not always be guaranteed to be the first data row, so this could result in flakiness.
3. Use my current technique of finding each specific csv row to verify by storage location name, and assert on those independently. (but this either results in multiple tests, or I could collapse it into a single test but then it would have multiple assertions - is that ok?)

Reply:
Sorry, just realized that extra storage location actually is created by the test, just not the new test(s) I added. It's created in the`before` block for `describe "GET #index"`. In this case, I think it could be modified to always have the same warehouse type, and then the csv test can have consistent data to verify.

Reply:
I've made the update to use a heredoc to verify generated csv contents in a single test.